File: backend/src/main/java/com/pawconnect/backend/user/dto/PublicUserResponse.java
package com.pawconnect.backend.user.dto;

import com.pawconnect.backend.dog.dto.DogResponse;
import com.pawconnect.backend.user.model.UserGender;
import lombok.Data;

import java.util.List;
import java.util.Set;

@Data
public class PublicUserResponse {
    private Long id;
    private String username;
    private String bio;
    private Integer age;
    private UserGender gender;
    private String profilePhotoUrl;
    private Set<String> languages;
    private List<DogResponse> dogs;
}

File: mobile/lib/src/models/public_user_response.dart
import 'dog_response.dart';

class PublicUserResponse {
  final int id;
  final String username;
  final String? bio;
  final int? age;
  final String? gender;
  final String? profilePhotoUrl;
  final Set<String> languages;
  final List<DogResponse> dogs;

  PublicUserResponse.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        username = json['username'],
        bio = json['bio'],
        age = json['age'],
        gender = json['gender'],
        profilePhotoUrl = json['profilePhotoUrl'],
        languages = Set<String>.from(json['languages'] ?? []),
        dogs = (json['dogs'] as List<dynamic>? ?? [])
            .map((e) => DogResponse.fromJson(e as Map<String, dynamic>))
            .toList();
}
